### PR TITLE
INTMDB-762: [CFN-PI] Automate Release Process CDK

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ INTMDB-762

This PR adds a GitHub action to automate the CDK release process: this action was **"inspired"** by the action used in CDKTF ([release.yml](https://github.com/cdktf/cdktf-provider-mongodbatlas/blob/main/.github/workflows/release.yml))

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] I have tested the CDK constructor in a CFN stack. See [TESTING.md](../TESTING.md)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        with:
+          node-version: 16.14.0
+      - name: Install dependencies
+        run: yarn install --check-files --frozen-lockfile
+      - name: release
+        run: npx projen release
+      - name: Backup artifact permissions
+        if: ${{ steps.git_remote.outputs.latest_commit == github.sha }}
+        run: cd dist && getfacl -R . > permissions-backup.acl
+        continue-on-error: true
```
Comment (colm-quinn) [inline .github/workflows/release.yml:34]: What is the backup used for here?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+      - name: Install Dependencies
+        run: cd .repo && yarn install --check-files --frozen-lockfile
+      - name: Create go artifact
+        run: cd .repo && npx projen package:go
+      - name: Collect go Artifact
+        run: mv .repo/dist dist
+      - name: Release
+        env:
+          GIT_USER_NAME: andreaangiolillo
+          GIT_USER_EMAIL: andrea.angiolillo@mongodb.com
```
Comment (colm-quinn) [inline .github/workflows/release.yml:264]: Leaking info - also this should be a service account

Comment (andreaangiolillo) [reply .github/workflows/release.yml:264]: oh thanks, good catch. `GIT_USER_NAME` and `GIT_USER_EMAIL` are not needed

Comment (andreaangiolillo) [reply .github/workflows/release.yml]: I will create a Jira ticket to create a GitHub service account. This will be likely used also for AtlasCLI releases

**Review by colm-quinn — APPROVED** [review]
LGTM, thanks for automating the CDK release process - its' great to have that in place!

Also, we should update/create a RELEASE.md to document the release process for the repo, so people with permissions can make a release if needed.

**Pull request merged**